crates/compiler/gen_llvm/src/llvm/compare.rs

```diff
@@ -1204,7 +1207,9 @@ fn build_tag_eq_help<'a, 'ctx>(
             let tags = other_tags;
             let mut cases = Vec::with_capacity_in(tags.len(), env.arena);
 
-            for (tag_id, field_layouts) in tags.iter().enumerate() {
+            for (i, field_layouts) in tags.iter().enumerate() {
+                let tag_id = if i >= (*nullable_id as _) { i + 1 } else { i };
+
```

Comment:
ugh. We should just get rid of this really just store the tag id in the lowest bits. `0x0` is just as invalid a pointer as `0x1`